src/NUnitFramework/framework/Internal/Execution/ParallelWorkItemDispatcher.cs

```diff
@@ -42,13 +42,19 @@ public class ParallelWorkItemDispatcher : IWorkItemDispatcher
         // Our Shifts
         private WorkShift _parallelShift = new WorkShift("Parallel");
         private WorkShift _nonParallelShift = new WorkShift("NonParallel");
+#if !NETCF
         private WorkShift _nonParallelSTAShift = new WorkShift("NonParallelSTA");
+#endif
 
         // Our Queues
         private WorkItemQueue _parallelQueue;
+#if !NETCF
         private WorkItemQueue _parallelSTAQueue;
+#endif
         private WorkItemQueue _nonParallelQueue;
+#if !NETCF
         private WorkItemQueue _nonParallelSTAQueue;
+#endif
```

Comment:
Instead of grouping shifts, nonparallel and parallel, you could just group by non-STA and STA